modules/platforms/dotnet/Apache.Ignite.Tests/Linq/LinqTests.cs

```diff
@@ -501,6 +522,102 @@ public void TestSelectDecimalIntoUserDefinedTypeUsesCorrectScale()
+        Assert.AreEqual(2, res[0].Int16);
+        Assert.AreEqual(3, res[0].Int32);
+        Assert.AreEqual(4, res[0].Int64);
+        Assert.AreEqual(5.5f, res[0].Float);
+        Assert.AreEqual(6.5, res[0].Double);
+        Assert.AreEqual(7.7, res[0].Decimal);
+        Assert.AreEqual(new LocalDate(2022, 12, 1), res[0].Date);
+        Assert.AreEqual(new LocalTime(11, 38, 1), res[0].Time);
+        Assert.AreEqual(new LocalDateTime(2022, 12, 19, 11, 1), res[0].DateTime);
+        Assert.AreEqual(Instant.FromUnixTimeSeconds(1), res[0].Timestamp);
+        Assert.AreEqual(new byte[] { 1, 2 }, res[0].Blob);
+    }
+
+    [Test]
+    public void TestSelectAllColumnTypesNullable()
+    {
+        var res = PocoAllColumnsSqlNullableView.AsQueryable()
+            .OrderBy(x => x.Key)
+            .ToList();
+
+        Assert.AreEqual(0, res[0].Key);
+        Assert.AreEqual(1, res[0].Int8);
+        Assert.AreEqual(2, res[0].Int16);
+        Assert.AreEqual(3, res[0].Int32);
+        Assert.AreEqual(4, res[0].Int64);
+        Assert.AreEqual(5.5f, res[0].Float);
+        Assert.AreEqual(6.5, res[0].Double);
+        Assert.AreEqual(7.7, res[0].Decimal);
+        Assert.AreEqual(new LocalDate(2022, 12, 1), res[0].Date);
+        Assert.AreEqual(new LocalTime(11, 38, 1), res[0].Time);
+        Assert.AreEqual(new LocalDateTime(2022, 12, 19, 11, 1), res[0].DateTime);
+        Assert.AreEqual(Instant.FromUnixTimeSeconds(1), res[0].Timestamp);
+        Assert.AreEqual(new byte[] { 1, 2 }, res[0].Blob);
+    }
+
+    [Test]
+    public void TestSelectAllColumnTypesNullableNull()
+    {
+        var res = PocoAllColumnsSqlNullableView.AsQueryable()
+            .OrderByDescending(x => x.Key)
+            .ToList();
+
+        Assert.AreEqual(100, res[0].Key);
+        Assert.IsNull(res[0].Str);
+        Assert.IsNull(res[0].Int8);
+        Assert.IsNull(res[0].Int16);
+        Assert.IsNull(res[0].Int32);
+        Assert.IsNull(res[0].Int64);
+        Assert.IsNull(res[0].Float);
+        Assert.IsNull(res[0].Double);
+        Assert.IsNull(res[0].Decimal);
+        Assert.IsNull(res[0].Date);
+        Assert.IsNull(res[0].Time);
+        Assert.IsNull(res[0].DateTime);
+        Assert.IsNull(res[0].Timestamp);
+        Assert.IsNull(res[0].Blob);
+    }
+
+    [Test]
+    public void TestWhereNull()
+    {
+        var query = PocoAllColumnsSqlNullableView.AsQueryable()
+            .Where(x => x.Int8 == null);
+
+        StringAssert.Contains("where (_T0.INT8 IS NOT DISTINCT FROM ?)", query.ToString());
+
+        var res = query.ToList();
+
+        Assert.AreEqual(100, res[0].Key);
+        Assert.AreEqual(1, res.Count);
+    }
+
+    [Test]
+    public void TestWhereNotNull()
+    {
+        var query = PocoAllColumnsSqlNullableView.AsQueryable()
+            .Where(x => x.Int8 != null)
+            .OrderBy(x => x.Key);
+
+        StringAssert.Contains("where (_T0.INT8 IS DISTINCT FROM ?)", query.ToString());
+
+        var res = query.ToList();
```

Comment:
Add another constraint? check that there are 10 items?

Reply:
Good point, fixed.